HW-25-CoreData/Source/CoreData/DataManager.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  DataManager.swift
+//  HW-25-CoreData
+//
+//  Created by Artem Muho on 21.05.2022.
+//
+
+import Foundation
+import CoreData
+import UIKit
+
+protocol DataMangerProtocol {
+    func getModels() -> [Friend]
+    func createFriend(name: String?, age: Date?, gender: String?)
+    func updateFriend(friend: Friend, name: String, age: Date, gender: String)
+    func deleteFriend(friend: Friend)
+}
+
+class DataManager: DataMangerProtocol {
+
+    private var context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+
+    func getModels() -> [Friend] {
+        var friendModel: [Friend] = []
+        do {
+            friendModel = try context.fetch(Friend.fetchRequest())
+        } catch { }
+        return friendModel
+    }
+
+    func createFriend(name: String?, age: Date?, gender: String?) {
+        let friend = Friend(context: context)
+        friend.name = name
+        friend.age = age
+        friend.gender = gender
+
+        do {
+            try context.save()
+        } catch {
+
+        }
+    }
+
+    func deleteFriend(friend: Friend) {
+        context.delete(friend)
+
+        do {
+            try context.save()
+        }
+        catch { }
+    }
+
+    func updateFriend(friend: Friend, name: String, age: Date, gender: String) {
+        friend.name = name
+        friend.age = age
+        friend.gender = gender
```

Comment:
Лишний код - сам в себя записываешь
```suggestion
    func updateFriend() {
```